packages/component-library-react/src/Checkbox.tsx

```diff
@@ -1,36 +1,64 @@
 import clsx from 'clsx';
-import { ForwardedRef, forwardRef, InputHTMLAttributes } from 'react';
+import { ForwardedRef, forwardRef, InputHTMLAttributes, useEffect, useImperativeHandle, useRef } from 'react';
 
 export interface CheckboxProps extends Omit<InputHTMLAttributes<HTMLInputElement>, 'type' | 'readOnly'> {
   appearance?: string;
+  indeterminate?: boolean;
   invalid?: boolean;
 }
 
 export const Checkbox = forwardRef(
   (
-    { appearance = 'custom', disabled, invalid, required, className, ...restProps }: CheckboxProps,
+    {
+      appearance = 'custom',
+      disabled,
+      indeterminate = false,
```

Comment:
I wonder if `undefined` wouldn't already be sufficient? Is an actual `false` value required for the indeterminate class name to work?